src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1125,18 +839,47 @@ private SSTableReader cloneAndReplace(DecoratedKey newFirst, OpenReason reason,
                                                  metadata,
                                                  ifile != null ? ifile.sharedCopy() : null,
                                                  dfile.sharedCopy(),
-                                                 newSummary,
                                                  bf.sharedCopy(),
                                                  maxDataAge,
                                                  sstableMetadata,
                                                  reason,
-                                                 header);
+                                                 header,
+                                                 newSummary);
+
         replacement.first = newFirst;
         replacement.last = last;
         replacement.isSuspect.set(isSuspect.get());
         return replacement;
     }
 
+    /**
+     * Clone this reader with the new values and set the clone as replacement.
+     *
+     * @param newBloomFilter for the replacement
+     *
+     * @return the cloned reader. That reader is set as a replacement by the method.
+     */
+    @VisibleForTesting
+    public SSTableReader cloneAndReplace(IFilter newBloomFilter)
+    {
+        SSTableReader replacement = internalOpen(descriptor,
+                                                 components,
+                                                 metadata,
+                                                 ifile != null ? ifile.sharedCopy() : null,
```

Comment:
@jasonstack Should `ifile` ever be `null` here? At least for this version of `cloneAndReplace()`, it looks like the only two things that use it (tests) pass without the null check...

Reply:
+1